packages/stream/solana/StreamClient.ts

```diff
@@ -890,6 +894,26 @@ export default class SolanaStreamClient extends BaseStreamClient {
     return sortedStreams.filter((stream) => stream[1].type === type);
   }
 
+  public async searchStreams(data: ISearchStreams): Promise<ProgramAccount<Stream>[]> {
+    const filters: MemcmpFilter[] = [];
+    for (const [k, v] of Object.entries(data)) {
+      if (v) {
+        filters.push({
+          memcmp: {
+            offset: STREAM_STRUCT_OFFSETS[k as keyof typeof STREAM_STRUCT_OFFSETS],
```

Comment:
related to the comment below about types - this thing could be undefined in rare cases